packages/aws_common/test/logging/aws_logger_test.dart

```diff
@@ -220,5 +222,76 @@ void main() {
       expect(loggerPlugin.timesCalled, 5);
       expect(globalLoggerPlugin.timesCalled, 0);
     });
+
+    test('Get parent returns parent in the logger hierarchy', () {
+      final logger = AWSLogger().createChild('child').createChild('grandChild');
+      expect(logger.parent, AWSLogger().createChild('child'));
+      expect(logger.parent?.parent, AWSLogger());
+      expect(logger.parent?.parent?.parent, isNull);
+    });
+
+    test('Get children returns children in the logger hierarchy', () {
+      final logger = AWSLogger();
+      final childLogger1 = logger.createChild('child-1');
+      final childLogger2 = logger.createChild('child-2');
+      final grandChildLogger = childLogger1.createChild('grandChild');
+
+      expect(logger.children.contains(childLogger1), isTrue);
+      expect(logger.children.contains(childLogger2), isTrue);
+      expect(childLogger1.children.contains(grandChildLogger), isTrue);
+      expect(grandChildLogger.children, isEmpty);
+    });
+
+    test('getPlugin returns registered plugin in the logger hierarchy', () {
+      final loggerPlugin = MockLoggerPlugin();
+      final logger = AWSLogger()..registerPlugin(loggerPlugin);
+      expect(logger.getPlugin<MockLoggerPlugin>(), loggerPlugin);
+
+      final childLogger = logger.createChild('child');
+      expect(childLogger.getPlugin<MockLoggerPlugin>(), loggerPlugin);
+
+      childLogger.unregisterPlugin(loggerPlugin);
+      expect(childLogger.getPlugin<MockLoggerPlugin>(), loggerPlugin);
+      expect(logger.getPlugin<MockLoggerPlugin>(), loggerPlugin);
```

Comment:
It's not clear to me what this is testing. Can you add a comment or use the `reason` parameter to `expect`?